libpod/image/image.go

```diff
@@ -149,43 +210,344 @@ func (i *Image) ID() string {
+	}
+	imgDigest := digest.Digest("")
+	if len(manifest) > 0 {
+		imgDigest = digest.Canonical.FromBytes(manifest)
+	}
+	annotations := annotations(manifest, manifestType)
+
+	ociv1Img, err := imgRef.OCIConfig()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting oci image info %q", i.ID())
+	}
+	info, err := imgRef.Inspect()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting image info %q", i.ID())
+	}
+
+	var repoDigests []string
+	for _, name := range i.Names() {
+		repoDigests = append(repoDigests, strings.SplitN(name, ":", 2)[0]+"@"+imgDigest.String())
+	}
+
+	data := &inspect.ImageData{
+		ID:              i.ID(),
+		RepoTags:        i.Names(),
+		RepoDigests:     repoDigests,
+		Comment:         ociv1Img.History[0].Comment,
+		Created:         ociv1Img.Created,
+		Author:          ociv1Img.Author,
+		Architecture:    ociv1Img.Architecture,
+		Os:              ociv1Img.OS,
+		ContainerConfig: &ociv1Img.Config,
+		Version:         info.DockerVersion,
+		Size:            size,
+		VirtualSize:     size + imgSize,
+		Annotations:     annotations,
+		Digest:          imgDigest,
+		Labels:          info.Labels,
+		RootFS: &inspect.RootFS{
+			Type:   ociv1Img.RootFS.Type,
+			Layers: ociv1Img.RootFS.DiffIDs,
+		},
+		GraphDriver: driver,
+	}
+	return data, nil
+}
+
+func annotations(manifest []byte, manifestType string) map[string]string {
+	annotations := make(map[string]string)
+	switch manifestType {
+	case ociv1.MediaTypeImageManifest:
+		var m ociv1.Manifest
+		if err := json.Unmarshal(manifest, &m); err == nil {
+			for k, v := range m.Annotations {
+				annotations[k] = v
+			}
+		}
+	}
+	return annotations
+}
+
+func (i *Image) toImageRef() (types.Image, error) {
+	ref, err := is.Transport.ParseStoreReference(i.imageruntime.store, "@"+i.ID())
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing reference to image %q", i.ID())
+	}
+	imgRef, err := ref.NewImage(nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading image %q", i.ID())
+	}
+	return imgRef, nil
+}
+
+// Decompose an Image
+func (i *Image) Decompose() error {
+	return types2.NotImplementedError
+}
+
+// TODO: Rework this method to not require an assembly of the fq name with transport
+/*
+// GetManifest tries to GET an images manifest, returns nil on success and err on failure
+func (i *Image) GetManifest() error {
+	pullRef, err := alltransports.ParseImageName(i.assembleFqNameTransport())
+	if err != nil {
+		return errors.Errorf("unable to parse '%s'", i.Names()[0])
+	}
+	imageSource, err := pullRef.NewImageSource(nil)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create new image source")
+	}
+	_, _, err = imageSource.GetManifest(nil)
+	if err == nil {
+		return nil
+	}
+	return err
+}
+*/
+
+// GetImage retrieves an image matching the given name or hash from system
+// storage
+// If no matching image can be found, an error is returned
+func (ir *ImageRuntime) GetImage(image string) (*Image, error) {
```

Comment:
If we have NewFromLocal do we need this?

Reply:
I'm thinking that we should make this private so we can still use it within the library, but people outside have to use NewFromLocal

Reply:
made private, good suggestion